pkg/performanceprofile/controller/performanceprofile/components/kubeletconfig/kubeletconfig.go

```diff
@@ -26,14 +26,20 @@ const (
 	// 4. Reserved CPUs
 	// 5. Memory manager policy
 	// Please avoid specifying them and use the relevant API to configure these parameters.
-	experimentalKubeletSnippetAnnotation = "kubeletconfig.experimental"
-	cpuManagerPolicyStatic               = "static"
-	cpuManagerPolicyOptionFullPCPUsOnly  = "full-pcpus-only"
-	memoryManagerPolicyStatic            = "Static"
-	defaultKubeReservedMemory            = "500Mi"
-	defaultSystemReservedMemory          = "500Mi"
-	defaultHardEvictionThreshold         = "100Mi"
-	evictionHardMemoryAvailable          = "memory.available"
+	experimentalKubeletSnippetAnnotation         = "kubeletconfig.experimental"
+	cpuManagerPolicyStatic                       = "static"
+	cpuManagerPolicyOptionFullPCPUsOnly          = "full-pcpus-only"
+	memoryManagerPolicyStatic                    = "Static"
+	defaultKubeReservedMemory                    = "500Mi"
+	defaultSystemReservedMemory                  = "500Mi"
+	defaultHardEvictionThresholdMemory           = "100Mi"
+	defaultHardEvictionThresholdNodefs           = "10%"
+	defaultHardEvictionThresholdImagefs          = "15%"
+	defaultHardEvictionThresholdNodefsInodesFree = "5%"
+	evictionHardMemoryAvailable                  = "memory.available"
+	evictionHardNodefsAvaialble                  = "nodefs.available"
+	evictionHardImagefsAvailable                 = "imagefs.available"
+	evictionHardNodefsInodesFree                 = "nodefs.inodesFree"
```

Comment:
I'm torn. On one hand, this is following the direction of the old code. On the other hand, we're duplicating k8s defaults, so this can get out of date, and most importantly, it can get out of date silently.
I think the minimum we should do is to add a test to ensure our defaults are still the same as k8s defaults.
The best option would be to read the defaults from the kubelet/k8s, but I'm not sure how practical it is.

Reply:
I hoped we would find out and document where the k8s defaults come from too. And yes, maintainability is the main concern here.

@marioferh Can you please add at least a comment close to the constants that will explain where the values came from? A k8s compliance test would be better and an automated way to read or update automatically the best. (Could go generate help here for example?)

Reply:
I have updated the vendor files to get defaults from kubelet files.